test/e2e/specs/fakerp/keyrotation.go

```diff
@@ -33,116 +24,50 @@ var _ = Describe("Key Rotation E2E tests [KeyRotation][Fake][LongRunning]", func
+		Expect(reflect.DeepEqual(before.Config.Certificates.EtcdCa.Cert, after.Config.Certificates.EtcdCa.Cert)).To(BeTrue())
+		Expect(reflect.DeepEqual(before.Config.Certificates.Ca.Cert, after.Config.Certificates.Ca.Cert)).To(BeTrue())
+
+		By("Verifying that certain non-ca public certificates have been updated")
+		Expect(reflect.DeepEqual(before.Config.Certificates.Admin.Cert, after.Config.Certificates.Admin.Cert)).To(BeFalse())
+		Expect(reflect.DeepEqual(before.Config.Certificates.AggregatorFrontProxy.Cert, after.Config.Certificates.AggregatorFrontProxy.Cert)).To(BeFalse())
+		Expect(reflect.DeepEqual(before.Config.Certificates.AzureClusterReader.Cert, after.Config.Certificates.AzureClusterReader.Cert)).To(BeFalse())
+		Expect(reflect.DeepEqual(before.Config.Certificates.EtcdClient.Cert, after.Config.Certificates.EtcdClient.Cert)).To(BeFalse())
+		Expect(reflect.DeepEqual(before.Config.Certificates.EtcdPeer.Cert, after.Config.Certificates.EtcdPeer.Cert)).To(BeFalse())
+		Expect(reflect.DeepEqual(before.Config.Certificates.EtcdServer.Cert, after.Config.Certificates.EtcdServer.Cert)).To(BeFalse())
+		Expect(reflect.DeepEqual(before.Config.Certificates.MasterKubeletClient.Cert, after.Config.Certificates.MasterKubeletClient.Cert)).To(BeFalse())
+		Expect(reflect.DeepEqual(before.Config.Certificates.MasterProxyClient.Cert, after.Config.Certificates.MasterProxyClient.Cert)).To(BeFalse())
+		Expect(reflect.DeepEqual(before.Config.Certificates.MasterServer.Cert, after.Config.Certificates.MasterServer.Cert)).To(BeFalse())
+		Expect(reflect.DeepEqual(before.Config.Certificates.NodeBootstrap.Cert, after.Config.Certificates.NodeBootstrap.Cert)).To(BeFalse())
+		Expect(reflect.DeepEqual(before.Config.Certificates.OpenShiftConsole.Cert, after.Config.Certificates.OpenShiftConsole.Cert)).To(BeFalse())
+		Expect(reflect.DeepEqual(before.Config.Certificates.OpenShiftMaster.Cert, after.Config.Certificates.OpenShiftMaster.Cert)).To(BeFalse())
+		Expect(reflect.DeepEqual(before.Config.Certificates.Registry.Cert, after.Config.Certificates.Registry.Cert)).To(BeFalse())
+		Expect(reflect.DeepEqual(before.Config.Certificates.Router.Cert, after.Config.Certificates.Router.Cert)).To(BeFalse())
+		Expect(reflect.DeepEqual(before.Config.Certificates.ServiceCatalogAPIClient.Cert, after.Config.Certificates.ServiceCatalogAPIClient.Cert)).To(BeFalse())
+		Expect(reflect.DeepEqual(before.Config.Certificates.ServiceCatalogServer.Cert, after.Config.Certificates.ServiceCatalogServer.Cert)).To(BeFalse())
+
+		// TODO: work towards moving the certs below to the updated section above
+		By("Verifying that certain non-ca public certificates have not been updated")
+		Expect(reflect.DeepEqual(before.Config.Certificates.GenevaLogging.Cert, after.Config.Certificates.GenevaLogging.Cert)).To(BeTrue())
+		Expect(reflect.DeepEqual(before.Config.Certificates.GenevaMetrics.Cert, after.Config.Certificates.GenevaMetrics.Cert)).To(BeTrue())
```

Comment:
You should also verify that nothing else has changed apart from the certificates.

Reply:
Unfortunately `InvalidSecrets` doesn't only update certificates but certain secrets as well. 

```
c.RegistryHTTPSecret = nil
c.RegistryConsoleOAuthSecret = ""
c.ConsoleOAuthSecret = ""
c.AlertManagerProxySessionSecret = nil
c.AlertsProxySessionSecret = nil
c.PrometheusProxySessionSecret = nil
c.SessionSecretAuth = nil
c.SessionSecretEnc = nil
c.Images.GenevaImagePullSecret = nil
```

Reply:
Right, but these fields are not accessible from the admin api. We probably need a different way to check that they actually get updated but I am not sure how this is going to look like for prod.

Reply:
I feel like we should be doing all these Expect()s in the unit test.  Here in the e2e we should be testing that the cluster still actually works.

Reply:
The healthcheck already runs as part of the rotate process. Anything more you would want to check? Maybe run e2e tests?

Reply:
we might want to identify secret relationship to E2E test suit and make sure we test them like this

Reply:
Should be simpler to run our e2es